From c96060d42ff44359faa92da021bb5b023ac53192 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Wed, 5 Sep 2012 18:21:35 -0400
Subject: [PATCH] greeter: explicitly quit when bus goes away

The process is supposed to get automatically terminated when
the session bus goes away, but that isn't seemingly working.

This commit forces it to quit explicitly.

https://bugzilla.gnome.org/show_bug.cgi?id=683068
---
 gui/simple-greeter/greeter-main.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/gui/simple-greeter/greeter-main.c b/gui/simple-greeter/greeter-main.c
index 85d9603..c52af96 100644
--- a/gui/simple-greeter/greeter-main.c
+++ b/gui/simple-greeter/greeter-main.c
@@ -129,32 +129,37 @@ static gboolean
 session_manager_connect (void)
 {
         GError *error;
 
         error = NULL;
 
         if (bus_connection == NULL) {
                 bus_connection = g_bus_get_sync (G_BUS_TYPE_SESSION, NULL, &error);
                 if (bus_connection == NULL) {
                         g_message ("Failed to connect to the session bus: %s",
                                    error->message);
                         g_error_free (error);
                         exit (1);
                 }
+
+                g_signal_connect (G_OBJECT (bus_connection),
+                                  "closed",
+                                  G_CALLBACK (gtk_main_quit),
+                                  NULL);
         }
 
         sm_proxy = gsm_manager_proxy_new_sync (bus_connection,
                                                G_DBUS_PROXY_FLAGS_NONE,
                                                SM_DBUS_NAME,
                                                SM_DBUS_PATH,
                                                NULL,
                                                &error);
 
         if (sm_proxy == NULL) {
                 g_message ("Failed to connect to the session manager: %s",
                            error->message);
                 g_error_free (error);
         }
 
         return (sm_proxy != NULL);
 }
